receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go

```diff
@@ -114,3 +115,8 @@ func getProcessHandlesInternal() (processHandles, error) {
 
 	return &gopsProcessHandles{handles: processes}, nil
 }
+
+func getProcessCreateTimeInternal(p processHandle) (uint64, error) {
+	v, e := p.CreateTime()
+	return uint64(v), e
+}
```

Comment:
`uint64` is still a not the format that is used eventually, right? We will need to get `pcommon.Timestamp` out of it anyway. Do we really need this wrapper? Can we use `p.CreateTime` instead?

Reply:
An `uint64` output makes the `s.scrapeProcessDelay.Milliseconds() > (time.Now().UnixMilli() - int64(createTime))` check simpler and makes mocking easier.

We need this wrapper for the `TestScrape` tests (these used to mock bootTime, but now they mock createTime) since these don't have easy access to the `p.CreateTime` method

Reply:
How does it make it easier? Doesn't `p.CreateTime` already return `int64` that you want in that line?

Reply:
Switched to using pcommon.Timestamp

Reply:
I was actually asking to remove the wrapper and use just `p.CreateTime` for simplicity and consistency with other stats getters that we have in the scraper

Reply:
We need this wrapper for the `TestScrape` tests (these used to mock bootTime, but now they mock createTime) since these don't have easy access to the `p.CreateTime` method. Here is the associated snippet

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4aaa26dea8d9eb639752e8904ec96d38d5898d31/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go#L81-L124

Reply:
I understand, but why do you need the wrapper? You can mock `p.CreateTime` which has `func(p processHandle) (uint64, error)` signature

Reply:
How would you mock `p.CreateTime` in the linked test case? I think it is not easily doable so perhaps I am missing something

Reply:
The same way as you mocking it currently, just change `pcommon.Timestamp` to `uint64`

```go
scraper.getProcessCreateTime = func(p processHandle) (uint64, error) { return createTime, nil } 
```

Reply:
It was originally set to `uint64` but was changed to pcommon.Timestamp due to the first comment in the chain. Do you want me to change it to how it was then? That is

```
func getProcessCreateTimeInternal(p processHandle) (uint64, error) {
	v, e := p.CreateTime()
	return uint64(v), e
}
```

Reply:
Or do you mean that it should match the createtime type exactly? That is 

```
func getProcessCreateTimeInternal(p processHandle) (int64, error) {
	v, e := p.CreateTime()
	return v, e
}
```

Reply:
You don't need any wrappers. I was assuming that p.CreateTime returns `(uint64, error)`. If it's `(int64, error)`, then the signature you need to mock is:

```go
func(p processHandle) (int64, error)
```

Reply:
```
func getProcessCreateTimeInternal(p processHandle) (int64, error) {
	return p.CreateTime()
}
```

is the same as

```go
var getProcessCreateTimeInternal = p.CreateTime
```

Reply:
> It was originally set to uint64 but was changed to pcommon.Timestamp due to the first comment in the chain.

My original comment was to remove the redundant wrapper

Reply:
Where would you insert the `var getProcessCreateTimeInternal = p.CreateTime` though? The `newProcessScraper` function doesn't have access to a `processHandle` `p`

Reply:
right, just use `processHandle.CreateTime`

Reply:
done

Reply:
Can you remove the wrapper now?